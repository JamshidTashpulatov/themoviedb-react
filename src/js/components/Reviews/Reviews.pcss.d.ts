export const container: string;
export const item: string;
export const author: string;
export const text: string;
