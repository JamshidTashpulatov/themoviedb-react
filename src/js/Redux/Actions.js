import apiConnect from "../services/ApiConnect";

export function getConfig() {
  return async (dispatch, getState) =>  {
    const response = await apiConnect.getConfig();
    dispatch({
      type: 'GET_CONFIG',
      config: response,
    });
  }
}

export function getLatestMovies() {
  return async (dispatch, getState) =>  {
    const response = await apiConnect.getMovies();
    dispatch({
      type: 'GET_LATEST_MOVIES',
      movies: response.results,
    });
  }
}

export function getGenres() {
  return async (dispatch, getState) =>  {
    const response = await apiConnect.getGenres();
    dispatch({
      type: 'GET_GENRES',
      genres: response,
    });
  }
}

export function getMovieFullview() {
  return async (dispatch, getState) =>  {
    const response = await apiConnect.getMovieFullview();
    dispatch({
      type: 'GET_MOVIE_FULL_VIEW',
      movieFullView: response,
    });
  }
}

export function getReviews() {
  return async (dispatch, getState) =>  {
    const response = await apiConnect.getReviews();
    dispatch({
      type: 'GET_REVIEWS',
      reviews: response,
    });
  }
}

export function getCredits() {
  return async (dispatch, getState) =>  {
    const response = await apiConnect.getCredits();
    dispatch({
      type: 'GET_CREDITS',
      credits: response,
    });
  }
}

export function getSimilar() {
  return async (dispatch, getState) =>  {
    const response = await apiConnect.getSimilar();
    dispatch({
      type: 'GET_SIMILAR',
      similar: response,
    });
  }
}

export function Search() {
  return async (dispatch, getState) =>  {
    const response = await apiConnect.Search();
    dispatch({
      type: 'SEARCH',
      search: response,
    });
  }
}

export function SearchByGenre() {
  return async (dispatch, getState) =>  {
    const response = await apiConnect.SearchByGenre();
    dispatch({
      type: 'SEARCH_BY_GENRE',
      searchByGenre: response,
    });
  }
}
