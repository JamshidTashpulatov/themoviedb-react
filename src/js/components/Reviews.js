import React from 'react';
import TextTruncate from 'react-text-truncate';
import Styles from '../../css/reviews.pcss';

const Reviews = (props) => {
  const NumReviews = 3;
  const reviews = props.data.results.slice(0, NumReviews).map(review => (
    <div className={Styles.item} key={review.id}>
      <h3 className={Styles.author}> {review.author}</h3>
      <p><TextTruncate containerClassName={Styles.content} line={10} truncateText="â€¦" text={review.content} /></p>
    </div>
  ));

  return (
    <div className="section section__dark">
      <div className="container">
        <h2 className="section--title">Reviews</h2>
        <div className={Styles.container}>
          {reviews}
        </div>
      </div>
    </div>
  );
}

Reviews.propTypes = {
  data: React.PropTypes.shape({
    results: React.PropTypes.array,
  }),
};

export default Reviews
