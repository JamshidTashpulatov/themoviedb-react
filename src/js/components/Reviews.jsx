import React from "react";
import { connect } from "react-redux";

import Styles from "../../css/reviews.pcss";

const Reviews = ({ reviews = [] }) => {
  const reviewsList = reviews.slice(0, 3).map(review => (
    <div className={Styles.item} key={review.id}>
      <h3 className={Styles.author}> {review.author}</h3>
      <p>
        {`${review.content.substr(0, 300)}...`}
      </p>
    </div>
  ));

  return <div className={Styles.container}>{reviewsList}</div>;
};

Reviews.propTypes = {
  reviews: React.PropTypes.array
};

const mapStateToProps = (state, ownProps) => {
  return {
    ...ownProps,
    reviews: state.movieFullView.reviews
  };
};

export default connect(mapStateToProps)(Reviews);
