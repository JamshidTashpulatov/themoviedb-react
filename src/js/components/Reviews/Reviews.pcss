.container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.item {
  flex-basis: 28%;
}

.author {
  text-align: center;
  margin-bottom: 0.5em;
  font-size: 18px;

  @media (--desktop) {
    font-size: 25px;
  }
}

.content {
  text-align: center;
  font-size: 12px;

  @media (--desktop) {
    font-size: 18px;
  }
}
