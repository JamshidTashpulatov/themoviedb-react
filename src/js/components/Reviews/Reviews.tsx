import * as React from "react";
import { connect } from "react-redux";
import Styles from "./Reviews.pcss";
type ReviewsProps = {
  reviews?: any[]
};
const Reviews: React.SFC<ReviewsProps> = ({ reviews = [] }) => {
  const reviewsList = reviews.slice(0, 3).map(review => (
    <div className={Styles.item} key={review.id}>
      <h3 className={Styles.author}> {review.author}</h3>
      <p className={Styles.text}>{`${review.content.substr(0, 300)}...`}</p>
    </div>
  ));
  return <div className={Styles.container}>{reviewsList}</div>;
};
const mapStateToProps = (state, ownProps) => {
  return {
    ...ownProps,
    reviews: state.movieFullView.reviews.data
  };
};
export default connect(mapStateToProps)(Reviews);
