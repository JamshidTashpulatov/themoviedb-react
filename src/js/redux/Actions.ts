import apiConnect from "../services/ApiConnect";
export function getConfig() {
  return async (dispatch, getState) => {
    const response = await apiConnect.getConfig();
    dispatch({
      type: "GET_CONFIG",
      config: response
    });
  };
}
export function getLatestMovies() {
  return async (dispatch, getState) => {
    const response = await apiConnect.getLatestMovies();
    dispatch({
      type: "GET_LATEST_MOVIES",
      movies: response.results
    });
  };
}
export function getMovieDetails(id) {
  return async (dispatch, getState) => {
    dispatch({ type: "GET_MOVIE_DETAILS_LOADING" });
    const response = await apiConnect.getMovieDetails(id);
    dispatch({
      type: "GET_MOVIE_DETAILS_DONE",
      details: response
    });
  };
}
export function getReviews(id) {
  return async (dispatch, getState) => {
    dispatch({ type: "GET_MOVIE_REVIEWS_LOADING" });
    const response = await apiConnect.getReviews(id);
    dispatch({
      type: "GET_MOVIE_REVIEWS_DONE",
      reviews: response.results
    });
  };
}
export function getCredits(id) {
  return async (dispatch, getState) => {
    dispatch({ type: "GET_MOVIE_CREDITS_LOADING" });
    const response = await apiConnect.getCredits(id);
    dispatch({
      type: "GET_MOVIE_CREDITS_DONE",
      credits: response
    });
  };
}
export function getSimilar(id) {
  return async (dispatch, getState) => {
    dispatch({ type: "GET_MOVIE_SIMILAR_LOADING" });
    const response = await apiConnect.getSimilar(id);
    dispatch({
      type: "GET_MOVIE_SIMILAR_DONE",
      similarMovies: response.results
    });
  };
}
export function getGenres() {
  return async (dispatch, getState) => {
    const response = await apiConnect.getGenres();
    dispatch({
      type: "GET_GENRES",
      genres: response
    });
  };
}
export function SearchByGenre(genre) {
  return async (dispatch, getState) => {
    const response = await apiConnect.SearchByGenre(genre);
    dispatch({
      type: "SEARCH_BY_GENRE",
      searchByGenre: response
    });
  };
}
export function Search(query) {
  return async (dispatch, getState) => {
    const response = await apiConnect.Search(query);
    dispatch({
      type: "SEARCH",
      search: response
    });
  };
}
