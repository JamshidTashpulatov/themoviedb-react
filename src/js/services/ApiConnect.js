/* eslint no-console: 0 */

import axios from "axios";

const baseUrl = "https://api.themoviedb.org/3/";
const configUrl = `${baseUrl}configuration?api_key=${process.env.API_KEY}`;
const moviesUrl = `${baseUrl}discover/movie?primary_release_year=2017&sort_by=popularity.desc&api_key=${process.env.API_KEY}`;

export default {
  getConfig: () =>
    axios
      .get(configUrl)
      .then(res => res.data)
      .catch(error => console.log(error)),

  getLatestMovies: () =>
    axios
      .get(moviesUrl)
      .then(res => res.data)
      .catch(error => console.log(error)),

  getGenres: () =>
    axios
      .get(`${baseUrl}genre/movie/list?api_key=${process.env.API_KEY}`)
      .then(res => res.data.genres)
      .catch(error => console.log(error)),

  getMovieDetails: id =>
    axios
      .get(`${baseUrl}movie/${id}?api_key=${process.env.API_KEY}`)
      .then(res => res.data)
      .catch(error => console.log(error)),

  getReviews: id =>
    axios
      .get(`${baseUrl}movie/${id}/reviews?api_key=${process.env.API_KEY}`)
      .then(res => res.data)
      .catch(error => console.log(error)),

  getCredits: id =>
    axios
      .get(`${baseUrl}movie/${id}/credits?api_key=${process.env.API_KEY}`)
      .then(res => res.data)
      .catch(error => console.log(error)),

  getSimilar: id =>
    axios
      .get(`${baseUrl}movie/${id}/similar?api_key=${process.env.API_KEY}`)
      .then(res => res.data)
      .catch(error => console.log(error)),

  Search: query =>
    axios
      .get(`${baseUrl}search/movie?api_key=${process.env.API_KEY}&query=${query}`)
      .then(res => res.data)
      .catch(error => console.log(error)),

  SearchByGenre: genre =>
    axios
      .get(`${baseUrl}discover/movie?api_key=${process.env.API_KEY}&with_genres=${genre}`)
      .then(res => res.data)
      .catch(error => console.log(error))
};
