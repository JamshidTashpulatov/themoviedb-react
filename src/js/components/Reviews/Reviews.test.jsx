import React from "react";
import { shallow } from "enzyme";
import Reviews from "./Reviews";
import { getReviews } from "../../redux/Actions";
//import { defaultState } from "../../redux/Store";
import { baseUrl } from "../../services/ApiConnect";
import configureStore from "redux-mock-store";
import nock from "nock";
import thunk from "redux-thunk";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const response = {
  id: 353486,
  page: 1,
  results: [
    {
      author: "piyushgupta69",
      content:
        "Previous 2 parts were much better story wise, however effects are good in this part.",
      id: "5a4ff969c3a3681bca00082a",
      url: "https://www.themoviedb.org/review/5a4ff969c3a3681bca00082a"
    },
    {
      author: "amanshroff",
      content:
        "**The 90's Game is back and got even more adventurous !!**\r\n\r\n\r\nWhen four teenagers in detention discover an old video game console with a game they've never heard of, they are immediately thrust into the game's jungle setting, into the bodies of their avatars, played by Dwayne Johnson, Jack Black, Kevin Hart, and Karen Gillan. What they discover is that you don't just play Jumanji -Jumanji plays you. They'll have to go on the most dangerous adventure of their lives, or they'll be stuck in the game forever.\r\n\r\n\r\n\r\nIn the age of Hollywood's obsession with franchise-building and reboots, the idea for a sequel to 1995's 'Jumanji' came across as another eye-roll worthy project. However, the combination of charming actors Dwayne Johnson, Jack Black, Karen Gillian and Kevin Hart, made it evident that this would wind up either being a complete trainwreck, or total mindless fun. Fortunately, 'Welcome to the Jungle' falls solidly into the latter category. The chemistry between the lead cast is evident and the script makes the most of this by giving each actor moments to shine and showcase their skills while they clearly enjoy playing their characters.\r\n\r\nJohnson's younger persona Spencer is a nerdy guy who has lived a sheltered life and is afraid of the world. When he enters the game he takes on the avatar of a hunk with no discernible physical weakness. Johnson has fun with this switch of persona and watching him go back and forth doesn't get old. The same goes for Jack Black who plays a self-absorbed, social media addicted teenage girl Bethany. Her switch into Black's trademark rotund shape is the most extreme and ends up being hilarious. While Black and Johnson are engaging as they are completely cast against their typical selves, Gillian and Hart aren't too far behind playing an awkward, insecure but intelligent girl, and an all-star jock boy trapped in a less-than-stellar body respectively.\r\n\r\n\r\n\r\nThis entire body-switch premise works on slapstick humour that's not crass and moves quickly with ample action to ensure that the 2-hour runtime doesn't feel bulky at any point. Granted there's a villain problem with Bobby Cannavale's Van Pelt being as one-dimensional as they come. It's conveniently explained away as this is all a video game, and the same rules apply for its lack of depth. \r\n\r\n\r\n\r\n_Overall_, 'Jumanji: Welcome To The Jungle' ends up being a fun, holiday film that will keep you entertained enough to make it worth your time and money.",
      id: "5a5c7031c3a3686d7c007302",
      url: "https://www.themoviedb.org/review/5a5c7031c3a3686d7c007302"
    },
    {
      author: "Gimly",
      content:
        "The effects have improved (as you'd hope they would over the course of 22 years), but _Welcome to the Jungle_ didn't really do much for me. It completely lacks the charm of the original movie, but also the worldly scope of the animated series. I sat there patiently as weak joke after weak joke bounced off of me ineffectually, but the person I watched it with seemed to be having a great time, so maybe I'm just dead inside?\r\n\r\n_Final rating:★★ - Definitely not for me, but I sort of get the appeal._",
      id: "5aa49b0dc3a3687c0c0197a7",
      url: "https://www.themoviedb.org/review/5aa49b0dc3a3687c0c0197a7"
    }
  ],
  total_pages: 1,
  total_results: 3
};

const request = nock(baseUrl)
  .get("/3/movie/123/reviews")
  .reply(200, response);

describe("reviews", () => {
  // @TODO: get initial state
  const store = mockStore({
    config: {
      images: {}
    },
    movies: {
      latest: []
    },
    genres: [],
    movieFullView: {
      details: {
        data: [],
        isLoading: false
      },
      similarMovies: {
        data: [],
        isLoading: false
      },
      reviews: {
        data: [],
        isLoading: false
      },
      credits: {
        data: [],
        isLoading: false
      }
    }
  });

  it("should dispatch correct actions", async () => {  
    await store.dispatch(getReviews(123));
    expect(store.getActions()).toMatchSnapshot(); 
  });

  it("renders correctly", () => {
    // @TODO: get reviews from state
    // const state = store.getState();
    // const reviews = state.movieFullView.reviews.data;
    const tree = shallow(<Reviews.WrappedComponent reviews={response.results} />);


    expect(tree).toMatchSnapshot();
  });
});
