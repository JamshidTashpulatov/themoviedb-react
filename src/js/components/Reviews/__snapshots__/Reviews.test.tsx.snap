// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reviews renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="item"
    key="5a4ff969c3a3681bca00082a"
  >
    <h3
      className="author"
    >
       
      piyushgupta69
    </h3>
    <p
      className="text"
    >
      Previous 2 parts were much better story wise, however effects are good in this part....
    </p>
  </div>
  <div
    className="item"
    key="5a5c7031c3a3686d7c007302"
  >
    <h3
      className="author"
    >
       
      amanshroff
    </h3>
    <p
      className="text"
    >
      **The 90's Game is back and got even more adventurous !!**


When four teenagers in detention discover an old video game console with a game they've never heard of, they are immediately thrust into the game's jungle setting, into the bodies of their avatars, played by Dwayne Johnson, Jack Black, ...
    </p>
  </div>
  <div
    className="item"
    key="5aa49b0dc3a3687c0c0197a7"
  >
    <h3
      className="author"
    >
       
      Gimly
    </h3>
    <p
      className="text"
    >
      The effects have improved (as you'd hope they would over the course of 22 years), but _Welcome to the Jungle_ didn't really do much for me. It completely lacks the charm of the original movie, but also the worldly scope of the animated series. I sat there patiently as weak joke after weak joke bounc...
    </p>
  </div>
</div>
`;

exports[`reviews should dispatch correct actions 1`] = `
Array [
  Object {
    "type": "GET_MOVIE_REVIEWS_LOADING",
  },
  Object {
    "reviews": Array [
      Object {
        "author": "piyushgupta69",
        "content": "Previous 2 parts were much better story wise, however effects are good in this part.",
        "id": "5a4ff969c3a3681bca00082a",
        "url": "https://www.themoviedb.org/review/5a4ff969c3a3681bca00082a",
      },
      Object {
        "author": "amanshroff",
        "content": "**The 90's Game is back and got even more adventurous !!**


When four teenagers in detention discover an old video game console with a game they've never heard of, they are immediately thrust into the game's jungle setting, into the bodies of their avatars, played by Dwayne Johnson, Jack Black, Kevin Hart, and Karen Gillan. What they discover is that you don't just play Jumanji -Jumanji plays you. They'll have to go on the most dangerous adventure of their lives, or they'll be stuck in the game forever.



In the age of Hollywood's obsession with franchise-building and reboots, the idea for a sequel to 1995's 'Jumanji' came across as another eye-roll worthy project. However, the combination of charming actors Dwayne Johnson, Jack Black, Karen Gillian and Kevin Hart, made it evident that this would wind up either being a complete trainwreck, or total mindless fun. Fortunately, 'Welcome to the Jungle' falls solidly into the latter category. The chemistry between the lead cast is evident and the script makes the most of this by giving each actor moments to shine and showcase their skills while they clearly enjoy playing their characters.

Johnson's younger persona Spencer is a nerdy guy who has lived a sheltered life and is afraid of the world. When he enters the game he takes on the avatar of a hunk with no discernible physical weakness. Johnson has fun with this switch of persona and watching him go back and forth doesn't get old. The same goes for Jack Black who plays a self-absorbed, social media addicted teenage girl Bethany. Her switch into Black's trademark rotund shape is the most extreme and ends up being hilarious. While Black and Johnson are engaging as they are completely cast against their typical selves, Gillian and Hart aren't too far behind playing an awkward, insecure but intelligent girl, and an all-star jock boy trapped in a less-than-stellar body respectively.



This entire body-switch premise works on slapstick humour that's not crass and moves quickly with ample action to ensure that the 2-hour runtime doesn't feel bulky at any point. Granted there's a villain problem with Bobby Cannavale's Van Pelt being as one-dimensional as they come. It's conveniently explained away as this is all a video game, and the same rules apply for its lack of depth. 



_Overall_, 'Jumanji: Welcome To The Jungle' ends up being a fun, holiday film that will keep you entertained enough to make it worth your time and money.",
        "id": "5a5c7031c3a3686d7c007302",
        "url": "https://www.themoviedb.org/review/5a5c7031c3a3686d7c007302",
      },
      Object {
        "author": "Gimly",
        "content": "The effects have improved (as you'd hope they would over the course of 22 years), but _Welcome to the Jungle_ didn't really do much for me. It completely lacks the charm of the original movie, but also the worldly scope of the animated series. I sat there patiently as weak joke after weak joke bounced off of me ineffectually, but the person I watched it with seemed to be having a great time, so maybe I'm just dead inside?

_Final rating:★★ - Definitely not for me, but I sort of get the appeal._",
        "id": "5aa49b0dc3a3687c0c0197a7",
        "url": "https://www.themoviedb.org/review/5aa49b0dc3a3687c0c0197a7",
      },
    ],
    "type": "GET_MOVIE_REVIEWS_DONE",
  },
]
`;
